zerver/tests/test_signup.py

```diff
@@ -3431,18 +3436,8 @@ def test_signup_without_password(self) -> None:
         Check if signing up without a password works properly when
         password_auth_enabled is False.
         """
-
-        email = self.nonreg_email("newuser")
-
-        result = self.client_post("/accounts/home/", {"email": email})
-        self.assertEqual(result.status_code, 302)
-        self.assertTrue(result["Location"].endswith(f"/accounts/send_confirm/{email}"))
-        result = self.client_get(result["Location"])
-        self.assert_in_response("Check your email so we can get started.", result)
-
-        # Visit the confirmation link.
-        confirmation_url = self.get_confirmation_url_from_outbox(email)
-        result = self.client_get(confirmation_url)
+        result = self.signup(email=self.nonreg_email("newuser"),
+                             stop_after_confirmation=True)
         self.assertEqual(result.status_code, 200)
 
         with patch("zerver.views.registration.password_auth_enabled", return_value=False):
```

Comment:
For this test, I think it's possible that just wrapping `self.verify_signup` in the `with ` statement would result in identical code, without needing to introduce `stop_after_confirmation`.  (I haven't tried to figure out if you'll need it for anything else).

Reply:
copied from my commit message:

> I had to pass stop_after_reg_form=True, as the call to get_user in
verify_signup fails. I am not sure whether this is the expected
behavior. Also this causes the test to use submit_reg_form_for_user,
meaning a blank password is posted to /accounts/register/ rather than
no password.

Reply:
Now trying to refactor `test_signup_email_message_contains_org_header`, I am getting the same issue with `get_user` not working. I suspect I am creating the user in a different realm? Is it the subdomain that determines that?

Reply:
Yes, the subdomain determines the realm where a user will be created, so you'll want to pass that into `get_user`.